subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/GeneratePluginDescriptors.java

```diff
@@ -58,11 +58,20 @@ public GeneratePluginDescriptors() {
         );
     }
 
+    /**
+     * Because of {@link PluginDeclaration} is not cacheable, the caching is handled by {@link #getImplementationClassById()}
+     */
     @Internal("The relevant data is tracked via implementationClassById")
     public ListProperty<PluginDeclaration> getDeclarations() {
         return declarations;
     }
 
+    /**
+     * Returns all {@code (id, implementation class)} pairs from {@link #getDeclarations()}.
+     * <p>
+     * This map will be responsible for guiding the input caching of this task,
+     * as {@link PluginDeclaration} is a non-cacheable object.
```

Comment:
```suggestion
     * This map is the only data needed from the plugin declarations to create the plugin descriptors.
```

How about this ☝️  ?

Reply:
Maybe an extra _input_, but close enough!
```suggestion
     * This map is the only input data needed from the plugin declarations to create the plugin descriptors.
```

Reply:
That is good too. You can also drop the `data` completely if you want to.